REFORGED/Undead/BuildSequence.ai

```diff
@@ -4,11 +4,16 @@
 function global_init_strategy takes nothing returns nothing
     call AddBlock(1, UNDEAD_ALTAR, true, 0, TOMB_OF_RELICS, 36)
     call AddBlock(1, UNDEAD_ALTAR, false, 0, CRYPT, 16)
-    call AddBlock(1, CRYPT, true, 0, TOMB_OF_RELICS, 36)
+    if towerrush then
+      call AddBlock(1, CRYPT, false, 0, TOMB_OF_RELICS, 20)
+    else
+      call AddBlock(1, CRYPT, true, 0, TOMB_OF_RELICS, 36)
+    endif
     call AddBlock(1, CRYPT, false, 0, GRAVEYARD, 20)
     call AddBlock(1, UNDEAD_ALTAR, false, 0, ZIGGURAT_1, 16)
     call AddBlock(1, UNDEAD_ALTAR, true, 0, ZIGGURAT_2, 20)
     call AddBlock(1, UNDEAD_ALTAR, true, 0, ZIGGURAT_FROST, 16)
+    call AddBlock(1, NECROPOLIS_2, false, 0, TOMB_OF_RELICS, 36)
```

Comment:
This is saying don't build a shop until necropolis_2 is under construction. That won't block the necropolis?

Reply:
I didn't notice the issue you mentioned , NECROPOLIS_1 not NECROPOLIS_2
This restriction is directly copied from the old version. If want to consider it more deeply, maybe better to limit the number of peon, because the restriction is to prevent the upgrade from starting before the number of peon is less than 5, which leads to slow development. This is obvious when there are double income and too many AI allies

Reply:
Yeah needs something different to solve anyway as this is blocking the TOMB_OF_RELICS from building instead.

Reply:
Actually, it's not true. The store's time is 36, and it's false here

Reply:
change code

Reply:
I think we can do something better. I believe if there is a requirement for more peons then we should block upgrade entirely.

If (peonCount < neededPeons (just pull out and call same logic from peonbuilder))
call AddBlock(neededPeons, racial_peon, true, 0, NECROPOLIS_2, 999999)
call AddBlock(neededPeons, racial_peon, true, 0, NECROPOLIS_1, 999999)
endif

And apply for every race. May be able to create a special block so we dont need to do that 99999, but if there more than one hall we also should skip the block.

Reply:
I don't understand why TIER1 needs to be restricted
There are not many scenarios that restrict workers here, so it seems unnecessary to add global variables

This restriction should only be sufficient for UD

And there are serious issues with the block
Basically, only the current quantity is judged here
There is no statistics on how much has been built cumulatively
If someone keeps slaughtering peon , resulting in a shortage of quantity, and many have actually been built, the time limit of 99999 is indeed unreasonable

Reply:
This solves the problem https://github.com/SMUnlimited/AMAI/commit/a7ba0ef4668e46a56d1c8477e61d38e0c7819d9e 
Avoids any peon control going into strategies and is only limited when its actually necessary, and you can still upgrade a town hall if its priority is set higher over peons.